src/Model/Traits/HasOtp.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Ecodev\Felix\Model\Traits;
+
+use Doctrine\ORM\Mapping as ORM;
+use Exception;
+use GraphQL\Doctrine\Annotation as API;
+use OTPHP;
+
+/**
+ * Trait for a user with second factor authentication using OTP.
+ */
+trait HasOtp
+{
+    /**
+     * @var bool
+     *
+     * @ORM\Column(type="boolean", options={"default" = 0})
+     */
+    private $otp = false;
+
+    /**
+     * @API\Exclude
+     *
+     * @ORM\Column(type="string", length=255, nullable=true)
+     */
+    private ?string $otpUri = null;
+
+    /**
+     * Must be implemented by the user class to use the login as the key label.
+     */
+    abstract public function getLogin(): ?string;
+
+    /**
+     * Enable 2FA for the user
+     * This should be only enabled after otpUri has been generated, stored and verified.
+     *
+     * @API\Exclude
+     */
+    public function setOtp(bool $otp): void
+    {
+        if ($otp === true && !empty($this->otpUri)) {
```

Comment:
```suggestion
        if ($otp && !empty($this->otpUri)) {
```

En plus de la simplification de booleen, est-ce que cela devrait être `!empty` ou `empty`  ?